pkg/cli/admin/release/extract.go

```diff
@@ -377,18 +377,24 @@ func (o *ExtractOptions) Run() error {
 						continue
 					}
 				}
-				credRequestManifests = append(credRequestManifests, m)
+				o.Manifests = append(o.Manifests, m)
 			}
-			if len(credRequestManifests) == 0 {
-				return true, nil
+			out := o.Out
+			if len(o.Directory) > 0 {
+				out, err = os.Create(path.Join(o.Directory, hdr.Name))
+				if err != nil {
+					return false, errors.Wrapf(err, "error creating manifest in %s", hdr.Name)
+				}
```

Comment:
Hey @soltysh , looks great... just one more suggestion (use filepath instead of path, include original error in return error message, and defer closing of the file so it doesn't stay open throughout the run):
```suggestion
				file, err := os.Create(filepath.Join(o.Directory, hdr.Name))
				if err != nil {
					return false, errors.Wrapf(err, "error creating manifest %s: %v", hdr.Name, err)
				}
				defer file.Close()
				out = file
```

Reply:
@csrwng yeah you're right, I always forget about it. Should be all addressed. thx!